src/Prismic/Api.php

```diff
@@ -42,70 +37,112 @@ class Api
     const EXPERIMENTS_COOKIE = "io.prismic.experiment";
 
     /**
-     * string the API's access token to be used with each API call
+     * The API's access token to be used with each API call
+     * @var string|null
      */
     protected $accessToken;
+
     /**
-     * ApiData the raw data of the /api document (prefer to use this class's instance methods)
+     * An instance of ApiData containing information about types, tags and refs etc
+     * @var ApiData
      */
     protected $data;
+
     /**
-     * CacheInterface the cache object specifying how to store the cache
+     * The cache instance
+     * @var CacheInterface
      */
     private $cache;
+
     /**
-     * Client
+     * Guzzle HTTP Client
+     * @var ClientInterface
      */
     private $httpClient;
 
-    /**
-     * Private constructor, not be used outside of this class.
-     */
     private function __construct(
-      $data /**< string */,
-      $accessToken = null /**< optional access token, if the API is private */,
-      Client $httpClient = null,
-      CacheInterface $cache = null)
-    {
+        ApiData $data,
+        ?string $accessToken = null,
+        ?ClientInterface $httpClient = null,
+        ?CacheInterface $cache = null
+    ) {
         $this->data        = $data;
         $this->accessToken = $accessToken;
-        $this->httpClient = is_null($httpClient) ? new Client() : $httpClient;
-        $this->cache = is_null($cache) ? self::defaultCache() : $cache;
+        $this->httpClient  = is_null($httpClient) ? new Client() : $httpClient;
+        $this->cache       = is_null($cache) ? self::defaultCache() : $cache;
+    }
+
+    /**
+     * This is the factory method with which to retrieve your API client instance
+     *
+     * If your API is set to "public" or "open", you can instantiate your Api object just like this:
+     * Api::get('https://your-repository-name.prismic.io/api/v2')
+     *
+     * @param  string          $action      The URL of your repository API's endpoint
+     * @param  string          $accessToken A permanent access token to use if your repository API is set to private
+     * @param  ClientInterface $httpClient  Custom Guzzle http client
+     * @param  CacheInterface  $cache       Cache implementation
+     * @param  int             $apiCacheTTL Max time to keep the API object in cache (in seconds)
+     * @return self
+     */
+    public static function get(
+        string            $action,
+        ?string           $accessToken = null,
+        ?ClientInterface  $httpClient = null,
+        ?CacheInterface   $cache = null,
+        int               $apiCacheTTL = 5
+    ) : self {
+        $cache    = is_null($cache) ? self::defaultCache() : $cache;
+        $cacheKey = $action . (empty($accessToken) ? "" : ("#" . $accessToken));
```

Comment:
the '?' might not work. in case of integration fields, you will have the '?' already with a query string parameter so you need to check that to put either a '?' or a '&'

Reply:
Are you referring yo L#103? If so, it would be better to `parse_url($action)` then `http_build_query` with the array of url params if there are potentially unknown query string parameters coming in to the named constructor?? What does the Api URL look like when using integration fields?

Reply:
with integration fields active, you'll have a url like <action_url>?integrationFieldsRef=123456

Reply:
and yep sorry l103, I went back there to comment, wrong line